index.go

```diff
@@ -175,6 +190,49 @@ func (u *UUIDFieldIndex) parseString(s string) ([]byte, error) {
 	return buf, nil
 }
 
+// parsePartialString parses a partial UUID from the string and returns its
+// value as a byte.
+func (u *UUIDFieldIndex) parsePartialString(s string) ([]byte, error) {
+	inputLength := len(s)
+	if inputLength == 0 {
+		return []byte{}, nil
+	}
+
+	parts := strings.Split(s, "-")
+	partsLength := len(parts)
+	if partsLength > 5 {
+		return nil, fmt.Errorf(`UUID should only have 4 "-" seperators`)
+	}
+
+	// Create the buffer. The sanatized length is the length of the original string
+	// without the "-" and the potentially necessary padding to make the input
+	// even.
+	sanatizedLength := inputLength - partsLength + 1 + (inputLength % 2)
+	buf := make([]byte, hex.DecodedLen(sanatizedLength))
+
+	// Decode each of the parts
+	index := 0
+	for _, part := range parts {
+		// Have to make the string even length. We append the 0 to the end
+		// because each part of full length UUID is even. Thus if we are at an
+		// odd length, there would be a next character to fill the lower order
+		// part of the byte. We are essentially creating the mask (pair & 0xf0)
```

Comment:
This won't really match the user's desired UUID though, right? They will have a different value for the last byte. I wonder if it's simpler just to throw an error if they give an odd number of characters since it doesn't meet the constraints of how the index works. If you do that you could just do `s = strings.Replace(s, "-", "", -1)` and then make sure that is even, otherwise throw an error.